### PR TITLE
Refactor Button and MenuItem components. Add Icon component. Remove FA default styles.

### PR DESCRIPTION
**Pull request by Oksamies**
Refactor Button component
Refactor MenuItem component
Add new Icon component
Disable default Fontawesome styles

refs TS-1826 and Adhoc decision to improve code quality

### CODE REVIEW

File: packages/cyberstorm/src/components/Layout/PackageDetailLayout/PackageDependencyList/PackageDependencyList.tsx [hunk 1]
```diff
@@ -81,7 +82,13 @@ export function PackageDependencyList(props: PackageDependencyListProps) {
             </div>
           </div>
         }
-        headerIcon={<FontAwesomeIcon icon={faBoxOpen} fixedWidth />}
+        headerIcon={
+          <Icon>
```
Comment (anttimaki) [inline packages/cyberstorm/src/components/Layout/PackageDetailLayout/PackageDependencyList/PackageDependencyList.tsx:86]: Yo dawg, we herd u like Icons in your Icons so you can wrap while you wrap.

File: packages/cyberstorm/src/components/Layout/PackageListLayout/PackageListLayout.tsx [hunk 2]
```diff
@@ -69,17 +74,24 @@ export function PackageListLayout(props: PackageListLayoutProps) {
...
-              icon={<FontAwesomeIcon icon={faDownload} fixedWidth />}
+              icon={
+                <Icon>
+                  <FontAwesomeIcon icon={faDownload} />
+                </Icon>
+              }
               colorScheme="accent"
               size="bold_large"
             />,
-            <Link
```
Comment (anttimaki) [inline packages/cyberstorm/src/components/Layout/PackageListLayout/PackageListLayout.tsx:76]: I would prefer to have some sort of explicit decision if we are to stop using the "anonymous" Link component for external links. Not just have it done "silently" as a part of huge refactoring PR.

File: packages/cyberstorm/src/components/Layout/PackageListLayout/PackageListLayout.tsx [hunk 3]
```diff
@@ -69,17 +74,24 @@ export function PackageListLayout(props: PackageListLayoutProps) {
...
-              key="meta-link"
-              leftIcon={<FontAwesomeIcon icon={faDiscord} fixedWidth />}
-              label="Join our community"
-              externalUrl="https://discord.thunderstore.io/"
-              size="bold"
-            />,
+            <a key="meta-link" href="https://discord.thunderstore.io/">
+              <Button.Root colorScheme="transparentPrimary">
```
Comment (anttimaki) [inline packages/cyberstorm/src/components/Layout/PackageListLayout/PackageListLayout.tsx:86]: Don't put button elements inside anchor elements.

File: packages/cyberstorm/src/components/Layout/Settings/Connections/Connections.tsx [hunk 4]
```diff
@@ -8,11 +8,13 @@ import styles from "./Connections.module.css";
...
+import { Icon } from "../../../Icon/Icon";
+import { OverwolfLogo } from "../../../../svg/svg";
 
 const PROVIDERS = [
-  { name: "Discord", icon: faDiscord },
-  { name: "GitHub", icon: faGithub },
-  { name: "Overwolf", icon: faGithub }, // TODO: icon
+  { name: "Discord", icon: <FontAwesomeIcon icon={faDiscord} /> },
+  { name: "GitHub", icon: <FontAwesomeIcon icon={faGithub} /> },
+  { name: "Overwolf", icon: OverwolfLogo() },
```
Comment (anttimaki) [inline packages/cyberstorm/src/components/Layout/Settings/Connections/Connections.tsx:17]: Could this be `<OverwolfLogo />` for consistency?

File: packages/cyberstorm/src/components/MenuItem/MenuItem.tsx [hunk 5]
```diff
@@ -3,10 +3,7 @@ import React, { ReactNode, useRef } from "react";
 import styles from "./MenuItem.module.css";
 
 export interface MenuItemProps {
-  label?: string;
-  leftIcon?: ReactNode;
-  rightIcon?: ReactNode;
-  colorScheme?: "default" | "accent";
+  children?: ReactNode | ReactNode[];
```
Comment (anttimaki) [inline packages/cyberstorm/src/components/MenuItem/MenuItem.tsx:6]: Just use `PropsWithChildren`?

File: packages/cyberstorm/src/components/MenuItem/MenuItem.tsx [hunk 6]
```diff
@@ -16,43 +13,19 @@ export const MenuItem: React.FC<MenuItemProps> = React.forwardRef<
...
 
 MenuItem.displayName = "MenuItem";
 
-const getStyle = (scheme: MenuItemProps["colorScheme"] = "default") => {
-  return {
-    default: "",
-    accent: "",
-  }[scheme];
-};
+const Root = MenuItem;
```
Comment (anttimaki) [inline packages/cyberstorm/src/components/MenuItem/MenuItem.tsx:30]: Export as.

File: packages/cyberstorm/src/components/MenuItem/MenuItemIcon.tsx [hunk 7]
```diff
@@ -0,0 +1,13 @@
+import { ReactNode } from "react";
+import styles from "./MenuItem.module.css";
+
+export interface MenuItemIconProps {
```
Comment (anttimaki) [inline packages/cyberstorm/src/components/MenuItem/MenuItemIcon.tsx:4]: No need to export? PropsWithChildren?

File: packages/cyberstorm/src/components/MetaItem/MetaItem.tsx [hunk 8]
```diff
@@ -26,7 +27,9 @@ export const MetaItem = React.forwardRef<HTMLDivElement, MetaItemProps>(
 
     return (
       <div ref={ref} className={`${styles.root} ${getStyle(colorScheme)}`}>
-        {icon}
+        <span className={styles.icon}>
```
Comment (anttimaki) [inline packages/cyberstorm/src/components/MetaItem/MetaItem.tsx:30]: The wrappers get rendered even if icon is undefined.

**Review by anttimaki — APPROVED** [review]
I only reviewed the changes in the code. Didn't think too much about how they affect things not included in the PR. Didn't view anything e.g. in storybook so I don't know they actually look like.

Lots of comments, of which at least some should be addressed. However I'm marking this PR as approved since it's *ridiculously* large and prone to merge conflicts. Address the issues in a separate PR.

File: packages/cyberstorm/src/components/Layout/Settings/Connections/Connections.module.css [hunk 9]
```diff
@@ -42,18 +42,18 @@
...
   display: flex;
   gap: var(--space--16);
   align-items: center;
   justify-content: center;
 }
 
+.connectionTypeInfoIcon {
+  width: 3rem;
+  height: 2.5rem;
+}
```
Comment (gkurck) [inline packages/cyberstorm/src/components/Layout/Settings/Connections/Connections.module.css:55]: I recommend setting the icon constraints as a square with 2.5rem to align everything nicely.

File: packages/cyberstorm/src/components/Layout/Settings/Connections/Connections.module.css [hunk 10]
```diff
@@ -42,18 +42,18 @@
...
 }
 
-.connectionTypeInfo > svg {
-  min-width: 40px;
-  min-height: 40px;
-}
-
 .connectionTypeInfo {
   display: flex;
   gap: var(--space--16);
```
Comment (gkurck) [inline packages/cyberstorm/src/components/Layout/Settings/Connections/Connections.module.css:47]: You can bump the gap to `--space--24` to take the comment below about the icon size into account.

**Review by gkurck — COMMENTED** [review]
I left a few comments.

No hard blockers as far as the scope of this PR goes.

There is the need for adding flex/centering to all icons which are set as squares to make sure they are centered and visual weight is retained despite the aspect ratio of a given icon. As per discussions, this and wrapping needs (full-w, fixed, inline) should probably be baked into the Icon component itself to reduce unnecessary repetition.

File: packages/cyberstorm/src/components/Button/Button.tsx [hunk 11]
```diff
@@ -7,19 +7,13 @@ import React, {
...
 import styles from "./Button.module.css";
 
-interface _ButtonProps extends _PlainButtonProps {
-  onClick?: MouseEventHandler<HTMLButtonElement>;
-  type?: "button" | "submit" | "reset";
-}
-
-interface _PlainButtonProps {
-  label?: string;
+export interface ButtonProps {
```
[the remark this answers is not in the document]
Comment (Oksamies) [reply packages/cyberstorm/src/components/Button/Button.tsx:10]: Nope, I'll clean up here on other places 👍

**Pull request merged**